### PR TITLE
Release v1.25.0

### PR DESCRIPTION
**Pull request by angrybayblade**
## Release summary

Version number: v1.25.0

## Release details

AEA:
- Fixes the mechanism to convert the `json` path the environment variable string
- Updates the process of agent subprocess termination to make sure we properly terminate agents across the various operating systems
- Introduces `reraise_as_click_exception` to re-raise exceptions as `click.ClickExceptions` on command definitions
- Extends `CliRunner` to allow usage of `capfd` to capture test output
- Introduce `generate_env_vars_recursively` method to auto generate the environment variable names for component overrides
- Extends `aea generate-key` to support creating multiple keys
- Extends the package manager API to
  - Update the hashes for third party packages with a warning
  - Update the dependency hashes when locking packages
  - Verifying the dependency hashes when verifying packages
- Adds deprecation warning for `aea hash all` command

Tests:
- Updates `libp2p` tests to use `capsys` to read `stdout` instead of patching `sys.stdout`
- Re enables tests skipped with `# need remote registry` comment
- Adds tests for package manager API

Chores:
- Updates the `tox` environment setting for unit tests to report duration of tests
- Deprecates the usage of `aea hash all` command from the workflow

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `main` branch (left side), from `develop`
- [ ] Lint and unit tests pass locally and in CI
- [ ] I have checked the fingerprint hashes are correct by running (`aea hash all` and `aea packages lock --check`)
- [ ] I have regenerated the latest API docs
- [ ] I built the documentation and updated it with the latest changes
- [ ] I have added an item in `HISTORY.md` for this release
- [ ] I bumped the version number in the `aea/__version__.py` file.
- [ ] I bumped the version number in every Docker image of the repo and published it. Also, I built and published them with tag `latest`  
      (check the READMEs of [`aea-develop`](../develop-image/README.md#publish) 
      and [`aea-user`](../user-image/README.md#publish))
- [ ] I have pushed the latest packages to the registry.
- [ ] I have uploaded the latest `aea` to PyPI.
- [ ] I have uploaded the latest plugins to PyPI.

### CODE REVIEW

File: HISTORY.md [hunk 1]
```diff
@@ -1,5 +1,29 @@
 # Release History - open AEA
 
+## 1.25.0 (2022-12-01)
+
+AEA:
+- Fixes the mechanism to convert the `json` path the environment variable string
+- Updates the process of agent subprocess termination to make sure we properly terminate agents across the various operating systems
+- Introduces `reraise_as_click_exception` to re-raise exceptions as `click.ClickExceptions` on command definitions
+- Extends `CliRunner` to allow usage of `capfd` to capture test output
+- Introduce `generate_env_vars_recursively` method to auto generate the environment variable names for component overrides
```
Comment (dvilelaf) [inline HISTORY.md:10]:
```suggestion
- Introduces `generate_env_vars_recursively` method to auto generate the environment variable names for component overrides
```

**Review by Karrenbelt — APPROVED** [review]

**Comment by Karrenbelt** [comment]
failures:
- ubuntu 3.8 - byte stream capture (not using `capfd` [here](https://github.com/valory-xyz/open-aea/blob/0905f9330ede6a6557058d25e96bd6a09a6da9fb/tests/test_cli/test_config.py#L55))
    ```
            result = self.runner.invoke(
                cli,
                [*CLI_LOG_OPTION, "config", "get", "skills.dummy.behaviours"],
                standalone_mode=False,
            )
            assert result.exit_code == 0
            
            >       actual_object = json.loads(result.output)
            ...
            >           raise JSONDecodeError("Expecting value", s, err.value) from None
    E           json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
    ```
- windows 3.7
    ```
    INTERNALERROR> MemoryError
    ```

**Comment by angrybayblade** [comment]
Ubuntu 3.8 Failure - stdout capture issue
Windows failure - memory error (We already have a prioritised issue for this)

File: Makefile [hunk 2]
```diff
@@ -71,8 +71,6 @@ test-sub-p:
 
 .PHONY: hashes
 hashes:
-	python -m aea.cli hash all
-	python -m aea.cli hash all --packages-dir=./tests/data/packages
 	python -m aea.cli packages lock
```
Comment (DavidMinarsch) [inline Makefile:74]: Does this now fully subsume above functionality? Needs to be in upgrading guide and release notes.

**Review by DavidMinarsch — CHANGES_REQUESTED** [review]

File: tox.ini [hunk 3]
```diff
@@ -190,13 +190,6 @@ skip_install = True
 deps =
 commands = {toxinidir}/scripts/check_copyright_notice.py --check
 
-[testenv:hash-all]
-skipsdist = True
-usedevelop = True
-commands =
-    pip install {toxinidir}[all]
-    aea hash all
-    aea hash all --packages-dir=./tests/data/packages
```
[the remark this answers is not in the document]
Comment (angrybayblade) [reply tox.ini:199]: `aea packages lock` will hash the package dependencies by default before updating the hash for the particular package

File: HISTORY.md [hunk 4]
```diff
@@ -1,5 +1,29 @@
 # Release History - open AEA
 
+## 1.25.0 (2022-12-01)
+
+AEA:
+- Fixes the mechanism to convert the `json` path the environment variable string
```
[the remark this answers is not in the document]
Comment (angrybayblade) [reply HISTORY.md]: Addressed [309db7a](https://github.com/valory-xyz/open-aea/pull/466/commits/309db7ad60ae4c071a0a2f26a33f377b035df95a)

File: HISTORY.md [hunk 5]
```diff
@@ -1,5 +1,29 @@
...
+- Updates the process of agent subprocess termination to make sure we properly terminate agents across the various operating systems
+- Introduces `reraise_as_click_exception` to re-raise exceptions as `click.ClickExceptions` on command definitions
+- Extends `CliRunner` to allow usage of `capfd` to capture test output
+- Introduces `generate_env_vars_recursively` method to auto generate the environment variable names for component overrides
+- Extends `aea generate-key` to support creating multiple keys
+- Extends the package manager API to
+  - Update the hashes for third party packages with a warning
+  - Update the dependency hashes when locking packages
+  - Verifying the dependency hashes when verifying packages
+- Adds deprecation warning for `aea hash all` command
```
[the remark this answers is not in the document]
Comment (angrybayblade) [reply HISTORY.md]: Yes, I've already added a deprecation warning on the `aea hash all` command, we haven't fully removed the command so I thought there was no point in adding a upgrade guide

[the remark this answers is not in the document]
Comment (angrybayblade) [reply HISTORY.md]: [309db7a](https://github.com/valory-xyz/open-aea/pull/466/commits/309db7ad60ae4c071a0a2f26a33f377b035df95a)

File: Makefile (hunk 2 above)
Comment (angrybayblade) [reply Makefile:74]: [309db7a](https://github.com/valory-xyz/open-aea/pull/466/commits/309db7ad60ae4c071a0a2f26a33f377b035df95a)

File: docs/upgrading.md [hunk 6]
```diff
@@ -7,6 +7,13 @@ Below we describe the additional manual steps required to upgrade between differ
 
 # Open AEA
 
+### Upgrade guide
+
+## `v1.24.0` to `v1.25.0`
+
+No backwards incompatible changes.
+
+Plugins from previous versions are not compatible anymore.
```
[the remark this answers is not in the document]
Comment (angrybayblade) [reply docs/upgrading.md:16]: [309db7a](https://github.com/valory-xyz/open-aea/pull/466/commits/309db7ad60ae4c071a0a2f26a33f377b035df95a)

**Review by DavidMinarsch — APPROVED** [review]

**Pull request merged**